src/ui/molecules/stripe-elements.svelte

```diff
@@ -194,120 +179,113 @@
       return;
     }
 
-    const { countryCode, postalCode } =
-      getCountryAndPostalCodeFromConfirmationToken(confirmationToken);
+    const billingAddress =
+      confirmationToken.payment_method_preview?.billing_details?.address;
+
+    const sameAddress =
+      billingAddress?.postal_code === lastTaxCustomerDetails?.postalCode &&
+      billingAddress?.country === lastTaxCustomerDetails?.countryCode;
 
-    if (
-      countryCode === lastTaxCustomerDetails?.countryCode &&
-      postalCode === lastTaxCustomerDetails?.postalCode
-    ) {
+    if (!billingAddress || sameAddress) {
       return;
     }
 
-    lastTaxCustomerDetails = { countryCode, postalCode };
+    lastTaxCustomerDetails = {
+      countryCode: billingAddress.country,
+      postalCode: billingAddress.postal_code,
+    } as TaxCustomerDetails;
 
-    onTaxCustomerDetailsUpdated({
-      countryCode,
-      postalCode,
-    });
+    onTaxCustomerDetailsUpdated(lastTaxCustomerDetails);
   }
 
-  function getCountryAndPostalCodeFromConfirmationToken(
-    confirmationToken: ConfirmationToken,
-  ): { countryCode?: string; postalCode?: string } {
-    const billingAddress =
-      confirmationToken.payment_method_preview?.billing_details?.address;
-    const countryCode = billingAddress?.country ?? undefined;
-    const postalCode = billingAddress?.postal_code ?? undefined;
-    return { countryCode, postalCode };
-  }
+  const onStripeElementsLoadingError = (error: any) => {
+    const actualError = error.error ? error.error : error;
+    onError({
+      code: PaymentElementErrorCode.ErrorLoadingStripe,
+      gatewayErrorCode: actualError.code ? actualError.code : undefined,
+      message: actualError.message,
+    });
+    onLoadingComplete();
+  };
 
-  onMount(() => {
-    updateStripeVariables();
+  const initStripe = async () => {
+    if (stripe) return;
+    if (elements) return;
+
+    try {
+      const { stripe: stripeInstance, elements: elementsInstance } =
+        await StripeService.initializeStripe(
+          gatewayParams,
+          brandingInfo,
+          stripeLocale,
+          stripeVariables,
+          viewport,
+        );
+      stripe = stripeInstance;
+      elements = elementsInstance;
 
+      // At this point, the gatewayParams might have been updated so we attempt to set them.
+      const elementsConfiguration = gatewayParams.elements_configuration;
+      if (elementsConfiguration) {
+        await StripeService.updateElementsConfiguration(
+          elements,
+          elementsConfiguration,
+        );
+      }
```

Comment:
I am not 100% sure but given that `initStripe` is called `onMount` and being awaited the comment does not seem right to me. Even more so since you already added the effect on `gatewayParams` changes this seems superfluous.

```suggestion
```